### DISCUSSION TITLE
301 redirect on next static export

### DISCUSSION THREAD

**Discussion by amanvlink** [vercel/next.js#53491] [Help]
### Summary

 I'd like to implement 301 redirects in the next static export. I have tried redirects in next.config.js and getServerSideProps but these methods are not supported in static export.
are there any other ways to implement 301 redirects?

**Comment by Nadeera3784** [comment]
HI @amanvlink,
How about using a custom middleware?
for example:

```
const redirectMiddleware = (req, res, next) => {
  if (req.url === '/old-url') {
    res.redirect('/new-url');
  } else {
    next();
  }
};

export default redirectMiddleware;
```

**Reply by amanvlink** [reply]
Hi @Nadeera3784 
middleware is not supported in static export.
https://nextjs.org/docs/pages/building-your-application/deploying/static-exports#unsupported-features
